node/cn/peer.go

```diff
@@ -993,28 +993,51 @@ func (p *multiChannelPeer) UpdateRWImplementationVersion() {
 
 func (p *multiChannelPeer) ReadMsg(rw p2p.MsgReadWriter, connectionOrder int, errCh chan<- error, wg *sync.WaitGroup, closed <-chan struct{}) {
 	defer wg.Done()
+
+	readMsgCh := make(chan struct {
+		p2p.Msg
+		error
+	}, channelSizePerPeer)
+	go func() {
+		for {
+			// TODO-klaytn: check 30-second timeout works
+			msg, err := rw.ReadMsg()
+			readMsgCh <- struct {
+				p2p.Msg
+				error
+			}{msg, err}
+		}
+	}()
```

Comment:
Will this goroutine be terminated?

Reply:
@jeongkyun-oh 
> Will this goroutine be terminated?

The child goroutine is expected to be terminated when the ReadMsg function is exited.
But to make sure, i changed the code for child goroutine to wait `closed`, also. If `closed` is close, the child goroutine can also receive that signal, and exit.
Can you take a look? https://github.com/klaytn/klaytn/pull/1389/commits/8f988f8458dc0ea3ed7c3d0a79de1eee65233605

> It looks better to launch a goroutine out of ReadMsg

Can I ask the reason why you think that launching this goroutine outside better? In my opinion, inside the ReadMsg looks better because the goroutine is dependent at ReadMsg.

Reply:
> The child goroutine is expected to be terminated when the ReadMsg function is exited.
But to make sure, i changed the code for child goroutine to wait closed, also. If closed is close, the child goroutine can also receive that signal, and exit.
Can you take a look? https://github.com/klaytn/klaytn/commit/8f988f8458dc0ea3ed7c3d0a79de1eee65233605

I think still the goroutine should be terminated before the main function terminated. But, it seems there is no panic or hang in the goroutine for now, so resolved this comment.

> Can I ask the reason why you think that launching this goroutine outside better? In my opinion, inside the ReadMsg looks better because the goroutine is dependent at ReadMsg.

Okay.